loky/backend/semaphore_tracker.py

```diff
@@ -81,25 +93,36 @@ def ensure_running(self):
                         args[i] = re.sub("-R+", "-R", args[i])
                 args += ['-c', cmd % r]
                 util.debug("launching Semaphore tracker: {}".format(args))
-                spawnv_passfds(exe, args, fds_to_pass)
-            except:
+                pid = spawnv_passfds(exe, args, fds_to_pass)
+            except BaseException:
                 os.close(w)
                 raise
             else:
                 self._fd = w
+                self._pid = pid
             finally:
                 os.close(r)
 
+    def _check_pid(self):
+        '''Check for the existence of the semaphore tracker process.'''
+        try:
+            self._send('PROBE', '')
+        except BrokenPipeError:
+            return False
+        else:
+            return True
+
     def register(self, name):
         '''Register name of semaphore with semaphore tracker.'''
+        self.ensure_running()
         self._send('REGISTER', name)
```

Comment:
It's weird that we send commands in text mode (that is unicode under Python 3) while we read the messages in binary mode `with os.fdopen(fd, 'rb') as f:` in `main`. But this refactoring should probably be done in a separate PR.

Reply:
The command is sent in byte mode. In the `_send` function, the msg is encoded in `ascii`.